src/analyzer/analyzerthread.h

```diff
@@ -0,0 +1,132 @@
+    WithBeats,
+    WithBeatsWithoutWaveform,
+};
+
+enum class AnalyzerThreadState {
+    Void,
+    Idle,
+    Busy,
+    Done,
+    Exit,
+};
+
+Q_DECLARE_TYPEINFO(AnalyzerThreadState, Q_MOVABLE_TYPE);
+Q_DECLARE_METATYPE(AnalyzerThreadState);
+
+// Atomic control values are used for transferring data between the
+// host and the worker thread, e.g. the next track to be analyzed or
+// the current analyzer progress that can be read independent of any
+// progress signal
+//
+// The frequency of progress signal is limited to avoid flooding the
+// signal queued connection between the internal worker thread and
+// the host, which might otherwise cause unresponsiveness of the host.
+class AnalyzerThread : public WorkerThread {
+    Q_OBJECT
+
+  public:
+    AnalyzerThread(
+            int id,
+            mixxx::DbConnectionPoolPtr pDbConnectionPool,
+            UserSettingsPointer pConfig,
+            AnalyzerMode mode = AnalyzerMode::Default);
+    ~AnalyzerThread() override = default;
+
+    int id() const {
+        return m_id;
+    }
+
+    // Submits the next track to the worker thread without
+    // blocking. This is only allowed after a progress() signal
+    // with state Idle has been received to avoid overwriting
+    // a previously sent track that has not been received by the
+    // worker thread, yet.
+    void submitNextTrack(TrackPointer nextTrack);
+
+  signals:
+    // Use a single signal for progress updates to ensure that all signals
+    // are queued and received in the same order as emitted from the internal
+    // worker thread. Different signals would otherwise end up in different
+    // queued connections which are processed in an undefined order!
+    // TODO(uklotzde): Encapsulate all signal parameters into an
+    // AnalyzerThreadProgress object and register it as a new meta type.
+    void progress(int threadId, AnalyzerThreadState threadState, TrackId trackId, AnalyzerProgress trackProgress);
+
+  protected:
+    void exec() override;
+
+    FetchWorkResult fetchWork() override;
+
+  private:
+    /////////////////////////////////////////////////////////////////////////
+    // Immutable values and pointers (objects are thread-safe)
+    const int m_id;
+    const mixxx::DbConnectionPoolPtr m_pDbConnectionPool;
+    const UserSettingsPointer m_pConfig;
+    const AnalyzerMode m_mode;
+
+    /////////////////////////////////////////////////////////////////////////
+    // Thread-safe atomic values
+
+    ControlValueAtomic<TrackPointer> m_nextTrack;
+
+    /////////////////////////////////////////////////////////////////////////
+    // Thread local: Only used in the constructor/destructor and within
+    // run() by the worker thread.
+
+    typedef std::unique_ptr<Analyzer> AnalyzerPtr;
+    std::vector<AnalyzerPtr> m_analyzers;
+
+    mixxx::SampleBuffer m_sampleBuffer;
+
+    TrackPointer m_currentTrack;
+
+    AnalyzerThreadState m_emittedState;
+
+    typedef std::chrono::steady_clock Clock;
```

Comment:
Just Clock can be everything in the using code. 
It feels also bad to trust this generic std:: implementation, without examine the properties, since the implementation may vary. Why not use the PerformanceTimer like elsewhere in Mixxx?